### ISSUE TITLE
Add spatial decomposition task

### ISSUE THREAD

**Issue by danielStrobl** [openproblems-bio/openproblems-v2#224]
### Task motivation

Spatial decomposition (also often referred to as Spatial deconvolution) is applicable to spatial transcriptomics data where the transcription profile of each capture location (spot, voxel, bead, etc.) do not share a bijective relationship with the cells in the tissue, i.e., multiple cells may contribute to the same capture location. The task of spatial decomposition then refers to estimating the composition of cell types/states that are present at each capture location.

### Proposed ground-truth in datasets

- [ ] DestVI[3](https://openproblems.bio/bibliography/#lopez2022destvi): scRNA-seq is generated based on learn NB parameters from the destVI manuscripts leveraging sparsePCA. Number of cells and cell types present in each spatial spot is computed via combination of kernel-based parametrization of a categorical distribution and the NB model.

- [ ] Pancreas (alpha=0.5)[11](https://openproblems.bio/bibliography/#luecken2022benchmarking): Human pancreas cells aggregated from single-cell (Dirichlet alpha=0.5).

- [ ] Pancreas (alpha=1)[11](https://openproblems.bio/bibliography/#luecken2022benchmarking): Human pancreas cells aggregated from single-cell (Dirichlet alpha=1).

- [ ] Pancreas (alpha=5)[11](https://openproblems.bio/bibliography/#luecken2022benchmarking): Human pancreas cells aggregated from single-cell (Dirichlet alpha=5).

- [ ] Tabula muris senis (alpha=0.5)[12](https://openproblems.bio/bibliography/#tabula2020single): Mouse lung cells aggregated from single-cell (Dirichlet alpha=0.5).

 - [ ] Tabula muris senis (alpha=1)[12](https://openproblems.bio/bibliography/#tabula2020single): Mouse lung cells aggregated from single-cell (Dirichlet alpha=1).

 - [ ] Tabula muris senis (alpha=5)[12](https://openproblems.bio/bibliography/#tabula2020single): Mouse lung cells aggregated from single-cell (Dirichlet alpha=5).

### Initial set of methods to implement

- [ ] Cell2location (alpha=20, amortised, hard-coded)[2](https://openproblems.bio/bibliography/#kleshchevnikov2022cell2location): Cell2location is a decomposition method based on Negative Binomial regression that is able to account for batch effects in estimating the single-cell gene expression signature used for the spatial decomposition step. Note that since batch information is unavailable in this task, here we use either a hard-coded reference, or a negative-binomial learned reference without batch labels. The parameter alpha refers to the detection efficiency prior. Links: [Docs](https://github.com/BayraktarLab/cell2location).

- [ ] Cell2location (alpha=1, reference hard-coded)[2](https://openproblems.bio/bibliography/#kleshchevnikov2022cell2location): Cell2location is a decomposition method based on Negative Binomial regression that is able to account for batch effects in estimating the single-cell gene expression signature used for the spatial decomposition step. Note that since batch information is unavailable in this task, here we use either a hard-coded reference, or a negative-binomial learned reference without batch labels. The parameter alpha refers to the detection efficiency prior. Links: [Docs](https://github.com/BayraktarLab/cell2location).

- [ ] Cell2location (alpha=20, reference hard-coded)[2](https://openproblems.bio/bibliography/#kleshchevnikov2022cell2location): Cell2location is a decomposition method based on Negative Binomial regression that is able to account for batch effects in estimating the single-cell gene expression signature used for the spatial decomposition step. Note that since batch information is unavailable in this task, here we use either a hard-coded reference, or a negative-binomial learned reference without batch labels. The parameter alpha refers to the detection efficiency prior. Links: [Docs](https://github.com/BayraktarLab/cell2location).

- [ ] Cell2location (alpha=200, reference hard-coded)[2](https://openproblems.bio/bibliography/#kleshchevnikov2022cell2location): Cell2location is a decomposition method based on Negative Binomial regression that is able to account for batch effects in estimating the single-cell gene expression signature used for the spatial decomposition step. Note that since batch information is unavailable in this task, here we use either a hard-coded reference, or a negative-binomial learned reference without batch labels. The parameter alpha refers to the detection efficiency prior. Links: [Docs](https://github.com/BayraktarLab/cell2location).

- [ ] Cell2location (alpha=20, NB reference)[2](https://openproblems.bio/bibliography/#kleshchevnikov2022cell2location): Cell2location is a decomposition method based on Negative Binomial regression that is able to account for batch effects in estimating the single-cell gene expression signature used for the spatial decomposition step. Note that since batch information is unavailable in this task, here we use either a hard-coded reference, or a negative-binomial learned reference without batch labels. The parameter alpha refers to the detection efficiency prior. Links: [Docs](https://github.com/BayraktarLab/cell2location).

- [ ] DestVI[3](https://openproblems.bio/bibliography/#lopez2022destvi): destVI is a decomposition method that leverages a conditional generative model of spatial transcriptomics down to the sub-cell-type variation level, which is then used to decompose the cell-type proportions determining the spatial organization of a tissue. Links: [Docs](https://github.com/YosefLab/scvi-tools).

- [ ] Non-Negative Matrix Factorization (NMF)[7](https://openproblems.bio/bibliography/#cichocki2009fast): NMF is a decomposition method based on Non-negative Matrix Factorization (NMF) that reconstructs expression of each spatial location as a weighted combination of cell-type signatures defined by scRNA-seq. It is a simpler baseline than NMFreg as it only performs the NMF step based on mean expression signatures of cell types, returning the weights loading of the NMF as (normalized) cell type proportions, without the regression step. Links: [Docs](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.NMF.html).

- [ ] NMF-reg[8](https://openproblems.bio/bibliography/#rodriques2019slide): NMFreg is a decomposition method based on Non-negative Matrix Factorization Regression (NMFreg) that reconstructs expression of each spatial location as a weighted combination of cell-type signatures defined by scRNA-seq. It was originally developed for Slide-seq data. Links: [Docs](https://github.com/tudaga/NMFreg_tutorial).

- [ ] Non-Negative Least Squares[4](https://openproblems.bio/bibliography/#aliee2021autogenes): NNLS13 is a decomposition method based on Non-Negative Least Square Regression (NNLS). It was originally introduced by the method AutoGenes. Links: [Docs](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.nnls.html).

- [ ] Random Proportions[13](https://openproblems.bio/bibliography/#openproblems): Random assignment of predicted celltype proportions from a Dirichlet distribution. Links: [Docs](https://github.com/openproblems-bio/openproblems).

- [ ] RCTD[5](https://openproblems.bio/bibliography/#cable2021robust): RCTD (Robust Cell Type Decomposition) is a decomposition method that uses signatures learnt from single-cell data to decompose spatial expression of tissues. It is able to platform effect normalization step, which normalizes the scRNA-seq cell type profiles to match the platform effects of the spatial transcriptomics dataset. Links: [Docs](https://github.com/dmcable/spacexr).

- [ ] SeuratV3[9](https://openproblems.bio/bibliography/#stuart2019comprehensive): SeuratV3 is a decomposition method that is based on Canonical Correlation Analysis (CCA). Links: [Docs](https://satijalab.org/seurat/archive/v3.2/spatial_vignette.html).

- [ ] Stereoscope[6](https://openproblems.bio/bibliography/#andersson2020single): Stereoscope is a decomposition method based on Negative Binomial regression. It is similar in scope and implementation to cell2location but less flexible to incorporate additional covariates such as batch effects and other type of experimental design annotations. Links: [Docs](https://github.com/scverse/scvi-tools).

- [ ] Tangram[10](https://openproblems.bio/bibliography/#biancalani2021deep): Tangram is a method to map gene expression signatures from scRNA-seq data to spatial data. It performs the cell type mapping by learning a similarity matrix between single-cell and spatial locations based on gene expression profiles. Links: [Docs](https://github.com/broadinstitute/Tangram).

- [ ] True Proportions[13](https://openproblems.bio/bibliography/#openproblems): Perfect assignment of predicted celltype proportions from the ground truth. Links: [Docs](https://github.com/openproblems-bio/openproblems).

### Proposed control methods

 - [ ] Random Proportions: Random assignment of predicted celltype proportions from a Dirichlet distribution. 
 - [ ] True Proportions: Perfect assignment of predicted celltype proportions from the ground truth.

### Proposed Metrics

- [ ] r2: R2, or the “coefficient of determination”, reports the fraction of the true proportion values’ variance that can be explained by the predicted proportion values. The best score, and upper bound, is 1.0. There is no fixed lower bound for the metric. The uniform/non-weighted average across all cell types/states is used to summarize performance.

**Comment by mumichae** [comment]
After looking at the task, it seems to me that it's a bit more complicated workflow (i.e. more than just dataset prep -> method -> metric), since you technically have 2 different inputs that need to be processed separately

**Comment by mumichae** [comment]
I added some suggestions on a workflow for spatial decomposition here: https://github.com/openproblems-bio/openproblems-v2/issues/292
